### PR TITLE
Update camera docs with new directory structure.

### PR DESCRIPTION
**Pull request by naushir**

### CODE REVIEW

**Comment by lurch** [comment]
ping @davidplowman Is deleting `libcamera_software.adoc` the correct approach here, or would it be better to add a link to it from somewhere?

File: documentation/asciidoc/computers/camera/libcamera_apps_building.adoc [hunk 1]
```diff
@@ -90,6 +90,8 @@ Users of Raspberry Pi OS can instead use
 meson build --buildtype=release -Dpipelines=raspberrypi -Dipas=raspberrypi -Dv4l2=true -Dgstreamer=enabled -Dtest=false -Dlc-compliance=disabled -Dcam=disabled -Dqcam=enabled -Ddocumentation=disabled
 ----
 
+NOTE: As of 06/05/2023, there has been a directory restructuring in the upstream libcamera git tree (https://git.libcamera.org/). If users are building libcamera from this tree, please replace `-Dpipelines=raspberrypi -Dipas=raspberrypi` with `-Dpipelines=rpi/vc4 -Dipas=rpi/vc4` in the above commands. This directory restructuring has not yet made it into the Raspberry Pi libcamera repo (https://github.com/raspberrypi/libcamera.git), so the above commands should be used without modifications if using that repo.
```
[the remark this answers is not in the document]
Comment (davidplowman) [reply documentation/asciidoc/computers/camera/libcamera_apps_building.adoc:93]: While we're editing this, should we add `-Dpycamera=enabled` to the command line?

**Comment by davidplowman** [comment]
> ping @davidplowman Is deleting `libcamera_software.adoc` the correct approach here, or would it be better to add a link to it from somewhere?

I presume it's an obsolete file now, so we should probably delete it?

File: documentation/asciidoc/computers/camera/libcamera_apps_building.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (naushir) [reply documentation/asciidoc/computers/camera/libcamera_apps_building.adoc:93]: I'll use 6th May 2023 here, keeping the commit date is probably handy.

[the remark this answers is not in the document]
Comment (naushir) [reply documentation/asciidoc/computers/camera/libcamera_apps_building.adoc:93]: I'll also add the -Dpycamera=enabled command line option.

**Review by lurch — APPROVED** [review]

**Comment by aallan** [comment]
> > ping @davidplowman Is deleting `libcamera_software.adoc` the correct approach here, or would it be better to add a link to it from somewhere?
> 
> I presume it's an obsolete file now, so we should probably delete it?

If it's not included from anywhere right now that would be my assumption. Is the content still needed (or replicated elsewhere?)

**Comment by naushir** [comment]
It's replicated in the main documentation page, I think it's just an orphaned file now.

**Comment by aallan** [comment]
Okay to kill it with fire then

**Pull request merged**